modules/gsp-namespace-role-policy-attachment/iam.tf

```diff
@@ -1,4 +1,4 @@
-resource "aws_iam_role_policy_attachment" "namespace-sqs" {
+resource "aws_iam_role_policy_attachment" "namespace-${var.role_name}" {
```

Comment:
Don't think this will work -don't think Terraform accepts dynamic resource names.